util/rate_limiter_test.cc

```diff
@@ -213,7 +213,7 @@ TEST_F(RateLimiterTest, AutoTuneIncreaseWhenFull) {
   // computes the next refill time (ensuring refill time in the future allows
   // the next request to drain the rate limiter).
   ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->SetCallBack(
-      "GenericRateLimiter::Refill", [&](void* /*arg*/) {
+      "GenericRateLimiter::Request:PostTimedWait", [&](void* /*arg*/) {
```

Comment:
I haven't gotten quite deep into how auto-tune works exactly so I need a quick context "refilling" in order to understand this change. 

Why we decide to change the call back position while still achieving the goals of "to advance (1) after Request()
has determined the bytes are not available; and (2) before Refill() computes the next refill time" mentioned in the comment above?

Reply:
Thank you for explaining what the new condition should be based on the impl change in this PR! 
> Unfortunately TimedWait() doesn't seem to advance the time in this mock time thing, though that may avoid the need for callback here entirely.

Feel free to either include the change to the callback (if the callback can be avoid) or open a task for me as part of the auto-tune learning process, IF some investigation into the issue "TimedWait() doesn't seem to advance the time in this mock time thing" is needed, not urgent and not blocking this PR.

If changing from "GenericRateLimiter::RefillBytesAndGrantRequests" to "GenericRateLimiter::Request:PostTimedWait" is enough, then I will really appreciate an update to the comment, since for a n00b like me who learns about auto-tune through reading this test case, unfortunately, can easily get hung up on the comment. 

Let me know how you would like to proceed on this!

Reply:
I only updated the comment. It looks like `TimedWait()` in tests (where we prefer to simulate wait rather than actually wait) is in a pretty confusing/inconsistent state so would take a long time to fix.

Reply:
Got you! I will keep that in mind when learning auto-tuning and reading the test case here.